src/components/App.tsx

```diff
@@ -481,6 +482,13 @@ class App extends React.Component<any, AppState> {
       this.initializeSocketClient({ showLoadingState: true });
     }
 
+    if (
+      this.state.editingLinearElement &&
+      !this.state.selectedElementIds[this.state.editingLinearElement.element.id]
+    ) {
+      this.actionManager.executeAction(actionFinalize);
+    }
+
```

Comment:
Not a fan of doing this in `componentDidUpdate`, but this is pretty much the only simple and predictable way to do it post-update. We need those Recoil selectors...